Ambit/Source/Ambit/Mode/ConfigImportExport.spec.cpp

```diff
@@ -1017,4 +1017,101 @@ void ConfigImportExportSpec::Define()
+    {
+        BeforeEach([this]()
+        {
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
+            check(AmbitMode);
+
+            // Set AWS S3 bucket to store test data
+            AmbitMode->UISettings->AwsRegion = "us-west-2";
+            AmbitMode->UISettings->S3BucketName = "ambit-unit-test-data";
+
+            // Set target platform for export
+            AmbitMode->UISettings->ExportPlatforms.SetWindows(true);
+
+            // Generate World
+            World = FAutomationEditorCommonUtils::CreateNewMap();
+
+            // Generate Exporter
+            Exporter = NewObject<UMockableConfigImportExport>();
+        });
+
+        It("Should Return FReply::Handled", [this]()
+        {
+            const FReply Reply = Exporter->OnExportMap();
+
+            TestTrue("Reply should be true on return", Reply.IsEventHandled());
+        });
+
+        AfterEach([this]()
+        {
+            GLevelEditorModeTools().DeactivateMode(FAmbitMode::EM_AmbitModeId);
+            Exporter = nullptr;
+            JsonContent = "";
+        });
+    });
+
+    Describe("OnExportGltf()", [this]()
+    {
+        BeforeEach([this]()
+        {
+            // Generate World
+            World = FAutomationEditorCommonUtils::CreateNewMap();
+
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
+            check(AmbitMode);
+
+            // Set AWS S3 bucket to store test data
+            AmbitMode->UISettings->AwsRegion = "us-west-2";
+            AmbitMode->UISettings->S3BucketName = "ambit-unit-test-data";
+
+            // Generate Exporter
+            Exporter = NewObject<UMockableConfigImportExport>();
+        });
+
+        It("Should fail if no mesh to export", [this]()
+        {
+            AddExpectedError("Cannot find a static mesh to export.", EAutomationExpectedErrorFlags::Exact, 1);
+            Exporter->OnExportGltf();
+        });
+
+        It("Should succeed after test map is loaded", [this]()
+        {
+            FAutomationEditorCommonUtils::LoadMap("/Ambit/Test/Maps/ProceduralPlacementTestMap");
+
+            // Create Mode Settings
+            GLevelEditorModeTools().ActivateMode(FAmbitMode::EM_AmbitModeId);
+
+            FAmbitMode* AmbitMode = FAmbitMode::GetEditorMode();
+            check(AmbitMode);
+
+            // Set AWS S3 bucket to store test data
+            AmbitMode->UISettings->AwsRegion = "us-west-2";
+            AmbitMode->UISettings->S3BucketName = "ambit-unit-test-data";
+
+            // Set target platform and file type for export
+            AmbitMode->UISettings->ExportPlatforms.SetWindows(true);
+            AmbitMode->UISettings->GltfType = "gltf";
+
+            const FReply Reply = Exporter->OnExportGltf();
```

Comment:
Is this doing an actual export to S3? If so, is there any chance of false positives or negatives if two people are running these tests at the same time? In general, at the unit test level you should avoid calling on the real service (S3 in this case) and use a mocked version of the service API instead. This improves test execution speed, eliminates intermittent failures due to service issues outside of your code base, and avoids race conditions and other conflicts when multiple people are running the same tests.

For now, I don't think we need to hold up the PR on this. I'd classify this as an improvement to make in the future and something to consider when writing new tests.

Reply:
This was originally doing an actual export to S3. The tests have now been modified to use mock S3 calls.